### ISSUE TITLE
twisted.internet.ssl.CertificateOptions docstring suggests server-only usage; CO is agnostic

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2014]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2014|
|Type|defect|
|Created|2006-08-23 13:31:07Z|

<details><summary>Searchable metadata</summary>

```
trac-id__2014 2014
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1156339867000000 1156339867000000
changetime__1161495572000000 1161495572000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

I guess this is all the change that is required:

```
Index: twisted/internet/_sslverify.py
===================================================================
--- twisted/internet/_sslverify.py	(revision 18490)
+++ twisted/internet/_sslverify.py	(working copy)
@@ -8,8 +8,8 @@
 from twisted.internet.defer import Deferred
 from twisted.internet.error import VerifyError, CertificateError
 
-# Private - shared between all ServerContextFactories, counts up to
-# provide a unique session id for each context
+# Private - shared between all OpenSSLCertificateOptions, counts up to provide
+# a unique session id for each context
 _sessionCounter = itertools.count().next
 
 class _SSLApplicationData(object):
@@ -798,8 +798,9 @@
 
 
 class OpenSSLCertificateOptions(object):
-    """A factory for SSL context objects, for server SSL connections.
     """
+    A factory for SSL context objects for both SSL servers and clients.
+    """
 
     _context = None
     # Older versions of PyOpenSSL didn't provide OP_ALL.  Fudge it here, just in case.
```

But maybe there are further documentation errors in this module which should be rectified?

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml set owner to @exarkun|
|-|-|

If there are further doc errors, then they can be filed under separate tickets. This patch looks good. Merge when ready.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [18499]) Apply patch from ssl.CO docstring ticket

Author: exarkun
Reviewer: jml
Fixes [#2014](https://github.com/twisted/twisted/issues/2014)

Remove documentation implying CertificateOptions is server-only.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
